app/controllers/job_invocations_controller.rb

```diff
@@ -0,0 +1,249 @@
+class JobInvocationsController < ApplicationController
+  def new
+    @composer = JobInvocationComposer.new(JobInvocation.new,
+                                          :host_ids => params[:host_ids],
+                                          :targeting => {
+                                            :targeting_type => Targeting::STATIC_TYPE,
```

Comment:
I don't think so, these are default attributes for new invocation (well we lose the ability to preselect targeting type, so we should fall back to static if params do not specify), when validation fails, you go through create action and only render `new` view.